src/engine/renderer/gl_shader.cpp

```diff
@@ -55,27 +67,226 @@ GLShader_motionblur                      *gl_motionblurShader = nullptr;
+	// We remove slash star xxxx star slash and
+	// slash slash xxxx to \r or \n combinations. We leave the
+	// new line character inplace to avoid splicing together lines
+	// and creating new tokens by accident.
+	// We don't use regex as we don't need it. Nor do we want
+	// to force a dependency on it either for such simple needs.
+	// We do not remove #if 0 type comments as buildshader does not do
+	// that. It would be also more complex. Complexity comes from
+	// things like #if X #elif 0 #else X #endif patterns.
+	// The shaders actually do/did that btw.
+	// But we don't need to handle #directives right now nor do we.
+	void StripComments(std::string& source)
+	{
+		size_t sourcePos = 0;
+		size_t keepPos = 0;
+		size_t commentStartPos;
+		size_t commentEndPos;
+		CommentType commentType;
+
+		auto keep = [&](size_t keepLength)
+		{
+			if (keepPos != sourcePos)
+				std::copy(source.begin() + sourcePos, source.begin() + sourcePos + keepLength,
+					source.begin() + keepPos);
+			keepPos += keepLength;
+		};
+
+		for (;; )
+		{
+			bool foundCommentStart = FindCommentStart(source, sourcePos, commentStartPos, commentType);
+			if (!foundCommentStart)
+			{
+				// If we don't find a comment, the rest of program is code.
+				size_t remainingLength = source.length() - sourcePos;
+				keep(remainingLength);
+				break;
+			}
+
+			bool foundCommentEnd = FindCommentEnd(source, commentStartPos, commentType, commentEndPos);
+			if (!foundCommentEnd)
+			{
+				// If we don't find the end of a comment:
+				// If it' a block comment we failed to close then keep everything
+				// before it and after. Before it is code and keeping what's
+				// after makes it easier to diagnose what's wrong.
+				// If we can't close a line comment (i.e. no new line), keep everything
+				// before it as that's code, but forget the line comment and what follows.
+				// Either way we are done.
+				if (commentType == CommentType::Block)
+				{
+					size_t remainingLength = source.length() - sourcePos;
+					keep(remainingLength);
+				}
+				else
+				{
+					size_t codeLength = commentStartPos - sourcePos;
+					keep(codeLength);
+				}
+				break;
+			}
+			// We found a start and end comment. keep the what's before the comment.
+			// as that's code. Then continue after the comment.
+			// Note that the new line that exists after comments is kept to avoid
+			// splicing any lines together without gaps that would create new unexpected tokens.
+			size_t codeLength = commentStartPos - sourcePos;
+			keep(codeLength);
+			sourcePos = commentEndPos;
+		}
+		source.resize(keepPos);
+	}
+
+	// Somewhat emulate what buildshader.sh does or doesn't do.
+	// Not wildly efficient but doesn't have to be as it's currently
+	// only used as part of a debugging only feature on small shader files.
+	void NormalizeShaderText( std::string& text )
 	{
-		delete _shaders[ i ];
+		size_t pos = 0;
+		while ((pos = text.find("\r\n", pos)) != std::string::npos)
```

Comment:
Ah I see, it is not exactly the same. However I think we can remove all \r characters as they are forbidden in other places in the shader: Feel free to change the script to do that same although I doubt a difference will ever arise.